drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -1022,4 +1077,26 @@ static int usb_rpi_init(void)
 	return 0;
 }
 
-SYS_INIT(usb_rpi_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+#define USB_RPI_PICO_PINCTRL_DT_INST_DEFINE(n)                                                     \
+	COND_CODE_1(DT_INST_PINCTRL_HAS_NAME(n, default), (PINCTRL_DT_INST_DEFINE(n)), ())
+
+#define USB_RPI_PICO_PINCTRL_DT_INST_DEV_CONFIG_GET(n)                                             \
+	COND_CODE_1(DT_INST_PINCTRL_HAS_NAME(n, default),                                          \
+		    ((void *)PINCTRL_DT_INST_DEV_CONFIG_GET(n)), (NULL))
+
+#define USB_DC_RPI_INIT(inst)                                                                      \
+	USB_RPI_PICO_PINCTRL_DT_INST_DEFINE(inst);                                                 \
+                                                                                                   \
+	static const struct udc_rpi_usbd_config udc_rpi_usbd_config##inst = {                      \
+		.pincfg = USB_RPI_PICO_PINCTRL_DT_INST_DEV_CONFIG_GET(inst),                       \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(inst, usb_rpi_init, NULL, NULL, &udc_rpi_usbd_config##inst,          \
+			      POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE, NULL);
+
+USB_DC_RPI_INIT(0);
+
+static inline const struct device *rpi_pico_usbd_device_get(void)
+{
+	return DEVICE_DT_INST_GET(0);
+}
```

Comment:
This does not make sense with this driver. And it is not that you were not pointed to a simpler solution. So please make the changes consistent, just add `const struct pinctrl_dev_config *pcfg`, put the DT macros above.

Reply:
Changed